File: UI/wwwroot/Questionnaires/FrontModel/Cat_Categorias_Test.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
//@ts-ignore
import { Tests } from './Tests.js';
class Cat_Categorias_Test extends EntityClass {
    /** @param {Partial<Cat_Categorias_Test>} [props] */
    constructor(props) {
        super(props, 'EntityQuestionnaires');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }
   /**@type {Number}*/ Id_categoria;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Imagen;
   /**@type {String}*/ Estado;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Array<Tests>} OneToMany*/ Tests;
}
export { Cat_Categorias_Test };

File: BusinessLogic/Questionnaires/Mapping/Model/Cat_Categorias_Test.cs
using APPCORE;
using APPCORE.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Cat_Categorias_Test : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_categoria { get; set; }
       public string? Descripcion { get; set; }
       public string? Imagen { get; set; }
       public string? Estado { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       [OneToMany(TableName = "Tests", KeyColumn = "Id_categoria", ForeignKeyColumn = "Id_categoria")]
       public List<Tests>? Tests { get; set; }
       
        public object? SaveCategoria()
        {
            Imagen = FileService.setImage(Imagen);
            return Save();
        }

        public object? UpdateCategoria()
        {
            Imagen = FileService.setImage(Imagen);
            return Update();
        }
   }
}
